contracts/HealthBlock.sol

```diff
@@ -172,10 +178,59 @@ contract HealthBlock {
 }
      function mapDoctorToProvider(address _providerAddress, address _doctorAddress) public{
         doctor storage d = doctors[_doctorAddress];
-       providerToDoctors[_providerAddress].push(doctor({name:d.name,email:d.email,specialization:d.specialization,id:_doctorAddress,age:d.age}));
+        uint8 idx =0;
+        bool found = false;
+        for(uint8 i =0;i< providerToDoctorRequests[_providerAddress].length;i++){
+            if(providerToDoctorRequests[_providerAddress][i].doctorAddr == _doctorAddress){
+                idx = i;
+                break;
+                found = true;
+ }
+        }
+           
```

Comment:
does this scale? just make sure this does not take time. a mapping is better positioned for such use cases.

Reply:
Thank you for the review. Addressing it in the next pr.